### PR TITLE
Review the 2 property rendering paths

### PR DESCRIPTION
**Pull request by AnthonySteele**
They differ in the checks, for no good reason.
Bring the safety of each to the other. Use common code.

### CODE REVIEW

File: src/NLog.StructuredLogging.Json/Helpers/Mapper.cs [hunk 1]
```diff
@@ -70,20 +70,28 @@ public static void HarvestToDictionary(IDictionary source, IDictionary<string, o
 
             foreach (DictionaryEntry property in source)
             {
-                HarvestToDictionary(property.Key.ToString(), property.Value, result, keyPrefixWhenCollision);
+                HarvestToDictionary(result, property.Key.ToString(), property.Value, keyPrefixWhenCollision);
             }
         }
 
-        private static void HarvestToDictionary(string key, object value, IDictionary<string, object> result, string keyPrefixWhenCollision)
+        private static void HarvestToDictionary(IDictionary<string, object> dest, string key, object value, string keyPrefixWhenCollision)
```
Comment (jplebre) [inline src/NLog.StructuredLogging.Json/Helpers/Mapper.cs:77]: result above, destination here?
Naming is a bit confusing

**Review by jplebre — COMMENTED** [review]
Test for the exception throwing on the flattenedlayout?

You mentioned you were removing common code but the duplication still there?
Not fussed about optimizing for a single repetition, just wondering if that's what you intended.

other than that LGTM

File: src/NLog.StructuredLogging.Json/Helpers/Mapper.cs (hunk 1 above)
Comment (AnthonySteele) [reply src/NLog.StructuredLogging.Json/Helpers/Mapper.cs:77]: I changed `result` -> `dest` as I thought that was a better name. And dest as the first param is possibly more conventional. Will change it in the other place as well.

**Comment by AnthonySteele** [comment]
There is still some duplication, yes. there are 2 Layout classes (see https://github.com/justeat/NLog.StructuredLogging.Json/issues/34) and the `try ... Render .. catch` can't be common between them as the types differ. But they both now have a `AddRenderedValue` method to make it clear that they're similar.

The code to put items in the dictionary with optional prefix was the common part (and already existed in the `Mapper`.
I'll be happy with this PR if the 2 Layouts are robust as each other.

**Comment by AnthonySteele** [comment]
Test added!

**Comment by jplebre** [comment]
looks good, will read it again tomorrow and merge it

File: src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+            var output = LogManager.Configuration.LogMessage(loggerName).First();
+
+            Assert.That(output, Does.StartWith(
+                "{\"duplicated\":\"value1\",\"attributes_duplicated\":\"value2\",\"TimeStamp\":\""));
+            Assert.That(output, Does.EndWith(
+                "\"prop1\":\"value1\",\"prop2\":\"2\"}"));
+        }
+
+        private void GivenLoggingIsConfiguredForTest(Target target)
```
Comment (jplebre) [inline src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs:54]: Should this and subsequent methods be moved to the `SimpleConfigurator`?

Don't mind either way

Comment (AnthonySteele) [reply src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs:54]: SimpleConfigurator is part of NLog, we don't change it.

**Pull request merged**